src/java/edu/stanford/math/plex4/graph/UndirectedWeightedListGraph.java

```diff
@@ -93,8 +95,10 @@ public boolean containsEdge(int i, int j) {
 	 * @see edu.stanford.math.plex_plus.graph.AbstractGraph#getNumEdges()
 	 */
 	public int getNumEdges() {
-		// TODO: complete
-		throw new UnsupportedOperationException();
+		int sum = 0;
+		for(int k : degrees.getValues())
+			sum += k;
+		return(sum/2);
```

Comment:
I think this is fine. In your example there are three edges: two self-edges, one at vertex 1 and one at vertex 3, and one edge between vertices 2 and 1.